crates/holochain_persistence_api/src/txn/mod.rs

```diff
@@ -0,0 +1,166 @@
+/// Transactional trait extensions to the CAS and EAV persistence
+use crate::{cas::content::*, cas::storage::ContentAddressableStorage, eav::*, error::*};
+use std::{
+    collections::BTreeSet,
+    marker::PhantomData,
+    sync::{Arc, RwLock},
+};
+
+/// Defines a transactional writer, typically implemented over a cursor.
+pub trait Writer {
+    /// Commits the transaction. Returns a `PersistenceError` if the
+    /// transaction does not succeed.
+    fn commit(&mut self) -> PersistenceResult<()>;
```

Comment:
I don't believe it should be possible to `commit` a writer twice, but with this signature

```
let writer = MyWriter::default();
writer.commit(); // takes &mut self
writer.commit(); // takes &mut self
```

```suggestion
    fn commit(self) -> PersistenceResult<()>;
```